metecho/api/views.py

```diff
@@ -199,6 +216,96 @@ class UserViewSet(mixins.RetrieveModelMixin, mixins.ListModelMixin, GenericViewS
     queryset = User.objects.all()
 
 
+class ProjectDependencyViewSet(ReadOnlyModelViewSet):
+    """Dependencies available during Project creation"""
+
+    permission_classes = (IsAuthenticated,)
+    serializer_class = ProjectDependencySerializer
+    queryset = ProjectDependency.objects.all()
+
+
+class GitHubOrganizationViewSet(ReadOnlyModelViewSet):
+    """GitHub Organizations available during Project creation"""
+
+    permission_classes = (IsAuthenticated,)
+    serializer_class = GitHubOrganizationSerializer
+    pagination_class = CustomPaginator
+    queryset = GitHubOrganization.objects.all()
+
+    @extend_schema(request=None, responses=ShortGitHubUserSerializer(many=True))
+    @action(detail=True, methods=["GET"], pagination_class=None)
+    def members(self, request, pk):
+        """Fetch the members of an Organization from GitHub"""
+        org: GitHubOrganization = self.get_object()
+        gh_api = gh.gh_as_user(request.user)
+        gh_org = gh_api.organization(org.login)
+        members = sorted(
+            (member.as_dict() for member in gh_org.members()),
+            key=lambda member: member["login"].lower(),
+        )
+        members = ShortGitHubUserSerializer(members, many=True)
+        return Response(members.data)
+
+    @extend_schema(
+        request=CheckRepoNameSerializer,
+        responses=OpenApiResponse(
+            {"type": "object", "properties": {"available": {"type": "boolean"}}},
+            description="",
+        ),
+    )
+    @action(detail=True, methods=["POST"])
+    def check_repo_name(self, request, pk):
+        """Determine if a repository name is available for the Organization on GitHub"""
+        org: GitHubOrganization = self.get_object()
+        serializer = CheckRepoNameSerializer(data=request.data)
+        serializer.is_valid(raise_exception=True)
+        try:
+            gh.get_repo_info(
+                None, repo_owner=org.login, repo_name=serializer.validated_data["name"]
+            )
+        except NotFoundError:
+            available = True
+        else:
+            available = False
+        return Response({"available": available})
+
+    @extend_schema(request=None, responses=GitHubAppInstallationCheckSerializer)
+    @action(detail=True, methods=["POST"])
+    def check_app_installation(self, request, pk):
```

Comment:
@jgerigmeyer I added a new endpoint that checks the following:

- User is part of the org
- The app is installed on the org
- The app has access to all repos
- The app has write permissions for repos and teams

I imagined this would be called when the user selects an org in the dropdown, but let me know if I should combine it with some other endpoint